diff -uNr a/torch/csrc/autograd/python_function.cpp b/torch/csrc/autograd/python_function.cpp
--- a/torch/csrc/autograd/python_function.cpp	2022-11-03 20:21:03.977859000 +0900
+++ b/torch/csrc/autograd/python_function.cpp	2022-11-04 16:21:44.196071311 +0900
@@ -497,7 +497,7 @@
       throw torch::TypeError(
           "save_for_backward can only save variables, but argument %ld is of "
           "type %s",
-          i,
+          (long)i,
           Py_TYPE(obj)->tp_name);
     }
   }
diff -uNr a/torch/csrc/utils/python_arg_parser.h b/torch/csrc/utils/python_arg_parser.h
--- a/torch/csrc/utils/python_arg_parser.h	2022-11-03 20:21:03.993860000 +0900
+++ b/torch/csrc/utils/python_arg_parser.h	2022-11-04 16:21:41.824173788 +0900
@@ -592,7 +592,7 @@
       args->signature.params[i].name.c_str(),
       args->signature.params[i].type_name().c_str(),
       Py_TYPE(obj)->tp_name,
-      idx + 1);
+      (long)(idx + 1));
 }
 
 inline std::vector<c10::SymInt> PythonArgs::symintlist(int i) {
@@ -753,7 +753,7 @@
           signature.params[i].name.c_str(),
           signature.params[i].type_name().c_str(),
           Py_TYPE(obj)->tp_name,
-          idx + 1);
+          (long)(idx + 1));
     }
   }
   return res;
